amethyst_renderer/src/cam.rs

```diff
@@ -76,6 +79,27 @@ impl Camera {
             std::f32::consts::FRAC_PI_3,
         ))
     }
+
+    /// Transforms position from screen space to camera space
+    pub fn position_from_screen(
+        &self,
+        screen_position: Vector2<f32>,
+        camera_transform: &GlobalTransform,
+        screen_dimensions: &ScreenDimensions,
+    ) -> Vector3<f32> {
+        let screen_x = 2.0 * screen_position.x / screen_dimensions.width() - 1.0;
+        let screen_y = 1.0 - 2.0 * screen_position.y / screen_dimensions.height();
+        let screen_point = Point3::new(screen_x, screen_y, 0.0).to_homogeneous();
+
+        let matrix = camera_transform.0
```

Comment:
Isn't this already a Vector3 or Point3?

Reply:
It's a `Matrix1x4`.
I haven't found an easy way to convert it to `Vector3` or `Point3`.

Reply:
Hm, I think it should be `Point3::from_homogeneous`

Reply:
Ah you are correct, as it's a 1x4 we can't just cast it. I think your initial solution is good then.